src/transformers/integrations/bitsandbytes.py

```diff
@@ -350,7 +350,7 @@ def dequantize_bnb_weight(weight: "torch.nn.Parameter", state=None):
         logger.warning_once(
             f"The model is going to be dequantized in {output_tensor.dtype} - if you want to upcast it to another dtype, make sure to pass the desired dtype when quantizing the model through `bnb_4bit_quant_type` argument of `BitsAndBytesConfig`"
         )
-        return output_tensor
+        return output_tensor.to(dtype)
```

Comment:
Why is this needed ? The dtype output should have been saved by the `QuantState`. Is it because this is needed for cpu since we can also load in fp32 / bf16 and most quantized model were quantized from the fp16 version ?

Reply:
Yes, exactly. The default data type is bf16 in CPU BNB.